packages/core/src/types.ts

```diff
@@ -386,12 +387,13 @@ export interface UserOperationStruct_v7 {
 //#endregion UserOperationStruct_v7
 
 //#region UserOperationStruct
-export type UserOperationStruct<TEntryPointVersion extends EntryPointVersion> =
-  TEntryPointVersion extends "0.6.0"
-    ? UserOperationStruct_v6
-    : TEntryPointVersion extends "0.7.0"
-    ? UserOperationStruct_v7
-    : never;
+export type UserOperationStruct<
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
+> = TEntryPointVersion extends "0.6.0"
+  ? UserOperationStruct_v6
+  : TEntryPointVersion extends "0.7.0"
+  ? UserOperationStruct_v7
+  : never;
```

Comment:
we should change this `never` to union type no? 

https://github.com/alchemyplatform/aa-sdk/pull/607/files#diff-887ed5859d7652b1d3596c31911d16babe9c65fe8b94ae1c92762907291791d7R73 would resolve as never no?

Reply:
hhmm, but it is already a union right? never will never hit.

UoRequest<EntryPointVersion> = v6 | v7

also UoRequest = v6 | v7